lib/Dialect/LLHD/IR/LLHDOps.cpp

```diff
@@ -64,6 +66,15 @@ void llhd::ConstantTimeOp::build(OpBuilder &builder, OperationState &result,
   return build(builder, result, TimeType::get(ctx), attr);
 }
 
+//===----------------------------------------------------------------------===//
+// SignalOp
+//===----------------------------------------------------------------------===//
+
+void SignalOp::getAsmResultNames(OpAsmSetValueNameFn setNameFn) {
+  if (getName() && !getName()->empty())
```

Comment:
I'm wondering why we don't actually verify that the string is not empty?

Reply:
I think there were some ops that used empty strings to encode an absent name... We should definitely change that to either null str attr or a non-empty string.